src/first_pass.c

```diff
@@ -42,77 +39,147 @@ boolean analyze_input_line(char* asm_line)
-					error_flag = ON;
-					break;
-				}
-				i++;
+				error_log("line %d: Invalid parameter for the instant address\n", line_counter);
+				result = FALSE;
+				break;
 			}
+			i++;
 		}
-		/*Immediate address*/
-		return '0';
 	}
-	/*Indirect register address*/
-	else if (*li == '*')
+	*addressing_mode = '0';
+	return result;
+}
+
+/**
+ * @brief Checks and sets the addressing mode for an indirect register address.
+ *
+ * This function checks if the given operand is a valid indirect register address (prefixed with `*r`).
+ * It validates that the register number is between 0 and 7. If the operand is valid, the addressing mode is set to `'2'`.
+ *
+ * @param li A pointer to the operand string.
+ * @param addressing_mode A pointer to a character where the addressing mode will be stored.
+ * @return A boolean value indicating whether the indirect register address is valid.
+ *         Returns TRUE if valid, FALSE otherwise.
+ */
+boolean indirect_register_address(char* li, char* addressing_mode) {
+	boolean result = TRUE;
+	if (!(*(li + 1) == 'r'))
 	{
-		if (!(*(li + 1) == 'r'))
-		{
-			printf("ERROR!! line %d: Invalid override parameter\n", line_counter);
-			error_flag = ON;
-		}
-		/*If the register is not between 0-7*/
-		else if (!(*(li + 2) > 47 && *(li + 2) < 56))
-		{
-			printf("ERROR!! line %d: Invalid indirect address registration\n", line_counter);
-			error_flag = ON;
-		}
-		return '2';
+		error_log("line %d: Invalid override parameter\n", line_counter);
+		result = FALSE;
 	}
-	/*Direct register address*/
-	else if (*li == 'r')
+	/*If the register is not between 0-7*/
+	if (!(*(li + 2) > 47 && *(li + 2) < 56))
 	{
-		/*If the register is not between 0-7*/
-		if (!(*(li + 1) > 47 && *(li + 1) < 56))
-		{
-			printf("ERROR!! line %d: Invalid direct address registration\n", line_counter);
-			error_flag = ON;
-		}
-		return '3';
+		error_log("line %d: Invalid indirect address registration\n", line_counter);
+		result = FALSE;
 	}
-	/*The operand is a label - Direct address*/
-	else if (*li > 20 && *li < 127)
-		return '1';
-	else
-		return ' ';
+	*addressing_mode = '2';
+	return result;
+}
+
+/**
+ * @brief Checks and sets the addressing mode for a direct register address.
+ *
+ * This function checks if the given operand is a valid direct register address (e.g., `r0`, `r1`, etc.).
+ * It validates that the register number is between 0 and 7. If the operand is valid, the addressing mode is set to `'3'`.
+ *
+ * @param li A pointer to the operand string.
+ * @param addressing_mode A pointer to a character where the addressing mode will be stored.
+ * @return A boolean value indicating whether the direct register address is valid.
+ *         Returns TRUE if valid, FALSE otherwise.
+ */
+boolean direct_register_address(char* li, char* addressing_mode) {
+	boolean result = TRUE;
+	/*If the register is not between 0-7*/
+	if (!(*(li + 1) > 47 && *(li + 1) < 56))
```

Comment:
Also here. I think I have a constant definition for 47 and 56.
If I don't have the use !(*(li + 2) > '0' && *(li + 2) < '7') for a cleaner code